docs/madr/decisions/010-timeout-policy.md

```diff
@@ -0,0 +1,174 @@
+Ad 2. Simpler config and you can still achieve the same result as previously by configuring it for `MeshSubset|MeshService|MeshServiceSubset`
+
+#### Connection and stream timeout
+I understand that `max connection timeout` and `max stream timeout ` are basically doing the same thing, since every 
+connection event in HTTP 1 are treated as streams in Envoy according to the docs.
+
+#### Route timeouts
+As for the route timeouts, we only care about `idle timeout` and `timeout` since `per try timeout` and `per try idle timeout` 
+are retry related timeouts and should be configured in retry policy.
+
+#### Request headers timeout
+Last one specific timeout is `request headers timeout` but this timeout makes no sense on outbound connections. 
+This only can be used to protect yourself from malicious inbound clients.
+
+#### Scaled timeouts
+Moreover, there is a mechanism of scaled timeout in envoy, which I am not sure if we want to expose to our users, at least not at first.
+
+### Specification
+Top-level targetRef can have all available kinds:
+```yaml
+targetRef:
+  kind: Mesh|MeshSubset|MeshService|MeshServiceSubset|MeshGatewayRoute|MeshHTTPRoute
+  name: ...
+```
+
+MeshUpstreamTimeout is outbound policy, that's why it has a "to" array:
+```yaml
+to:
+  - targetRef:
+      kind: Mesh|MeshSubset|MeshService|MeshServiceSubset|MeshHTTPRoute
+      name: ...
+```
+
+#### Minimal configuration
+Taking into account previous considerations, the minimal configuration for `MeshUpstreamTimeout` policy should look like this:
+```yaml
+default:
+  connectTimeout: ... # all protocols
+  idleTimeout: ... # all protocols
+  http:
+    requestTimeout: ... 
+    streamIdleTimeout: ...
+    maxStreamDuration: ...
+```
+
+
+### Examples
+#### Default configuration
+```yaml
+type: MeshUpstreamTimeout  
+mesh: default  
+name: default-timeouts  
+spec:  
+  targetRef:  
+    kind: Mesh  
+    name: default  
+  to:  
+    - targetRef:  
+        kind: Mesh  
+        name: default  
+      default:  
+        connectTimeout: 5s  
+        idleTimeout: 1h  
+        http:  
+          requestTimeout: 15s  
+          streamIdleTimeout: 30m  
+          maxStreamDuration: 0s
+```
+
+#### Configuration with overridden TCP specific timeouts for database
+
+```yaml
+type: MeshUpstreamTimeout  
+mesh: default  
+name: default-timeouts  
+spec:  
+  targetRef:  
+    kind: Mesh  
+    name: default  
+  to:  
+    - targetRef:  
+        kind: Mesh  
+        name: default  
+      default:  
+        connectTimeout: 5s  
+        idleTimeout: 1h  
+        http:  
+          requestTimeout: 15s  
+          streamIdleTimeout: 30m  
+          maxStreamDuration: 0s  
+---  
+type: MeshUpstreamTimeout  
+mesh: default  
+name: database-tcp-timeouts  
+spec:  
+  targetRef:  
+    kind: Mesh  
+    name: default  
+  to:  
+    - targetRef:  
+        kind: MeshService  
+        name: postgres  
+      default:  
+        connectTimeout: 1s  
+        idleTimeout: 10m
+```
```

Comment:
Can we have an example for Gateway? Can we even put MeshGatewayRoute in top-level targetRef or is it MeshService in top-level and MeshGatewayRoute in `to` section? If needed, ask @michaelbeaumont for help here

Reply:
still missing the example

Reply:
Added example. I think we can use `MeshGatewayRoute` in top level targetRef as we use it in `MeshTrafficPermission` policy. But I don't think we can put `MeshGatewayRoute` in  to section